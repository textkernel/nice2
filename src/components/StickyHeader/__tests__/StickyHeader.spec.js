import React from 'react';
import toJson from 'enzyme-to-json';
import { StickyHeader } from '..';

describe('StickyHeader component', () => {
    let wrapper;
    let consoleError;

    beforeEach(() => {
        consoleError = jest.spyOn(console, 'error').mockImplementationOnce(() => {});
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render StickyHeader correctly', () => {
        wrapper = shallow(
            <StickyHeader>
                <div>I am a header</div>
            </StickyHeader>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(consoleError).not.toHaveBeenCalled();
    });

    it('should render StickyHeader with headerClassName', () => {
        wrapper = shallow(
            <StickyHeader className="my-sticky-class">
                <div>I am a header</div>
            </StickyHeader>
        );
        expect(wrapper.find('.StickyHeader').hasClass('my-sticky-class')).toBe(true);
    });
});
