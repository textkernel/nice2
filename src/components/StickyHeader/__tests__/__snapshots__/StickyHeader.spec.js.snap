// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickyHeader component should render StickyHeader correctly 1`] = `
<div
  className="StickyHeader"
>
  <div>
    I am a header
  </div>
</div>
`;
